pynestml/codegeneration/nest_code_generator.py

```diff
@@ -173,31 +173,26 @@ def __init__(self, options: Optional[Mapping[str, Any]] = None):
         self.setup_printers()
 
     def run_nest_target_specific_cocos(self, neurons: Sequence[ASTModel], synapses: Sequence[ASTModel]):
-        for neuron in neurons:
-            # Check if the random number functions are used only in the update block.
-            CoCosManager.check_co_co_nest_random_functions_legally_used(neuron)
-            if Logger.has_errors(neuron):
-                raise Exception("Error(s) occurred during code generation")
-
-        for synapse in synapses:
-            # Check if the random number functions are used only in the update block.
-            CoCosManager.check_co_co_nest_random_functions_legally_used(synapse)
-
-            synapse_name_stripped = removesuffix(removesuffix(synapse.name.split("_with_")[0], "_"), FrontendConfiguration.suffix)
-
-            # special case for NEST delay variable (state or parameter)
-            assert synapse_name_stripped in self.get_option("delay_variable").keys(), "Please specify a delay variable for synapse '" + synapse_name_stripped + "' in the code generator options (see https://nestml.readthedocs.io/en/latest/running/running_nest.html#dendritic-delay-and-synaptic-weight)"
-            assert ASTUtils.get_variable_by_name(synapse, self.get_option("delay_variable")[synapse_name_stripped]), "Delay variable '" + self.get_option("delay_variable")[synapse_name_stripped] + "' not found in synapse '" + synapse_name_stripped + "' (see https://nestml.readthedocs.io/en/latest/running/running_nest.html#dendritic-delay-and-synaptic-weight)"
-
-            # special case for NEST weight variable (state or parameter)
-            assert synapse_name_stripped in self.get_option("weight_variable").keys(), "Please specify a weight variable for synapse '" + synapse_name_stripped + "' in the code generator options (see https://nestml.readthedocs.io/en/latest/running/running_nest.html#dendritic-delay-and-synaptic-weight)"
-            assert ASTUtils.get_variable_by_name(synapse, self.get_option("weight_variable")[synapse_name_stripped]), "Weight variable '" + self.get_option("weight_variable")[synapse_name_stripped] + "' not found in synapse '" + synapse_name_stripped + "' (see https://nestml.readthedocs.io/en/latest/running/running_nest.html#dendritic-delay-and-synaptic-weight)"
-
-            if self.option_exists("delay_variable") and synapse_name_stripped in self.get_option("delay_variable").keys():
-                delay_variable = self.get_option("delay_variable")[synapse_name_stripped]
-                CoCoNESTSynapseDelayNotAssignedTo.check_co_co(delay_variable, synapse)
-
-            if Logger.has_errors(synapse):
+        for model in neurons + synapses:
+            # Check if the random number functions are used in the right blocks
+            CoCosManager.check_co_co_nest_random_functions_legally_used(model)
+
+            if self.get_option("neuron_synapse_pairs"):
```

Comment:
should this only run for synapse models though? I don't understand how this is passing CI if line 181 is run on a neuron model rather than a synapse model.

Reply:
The CoCo `check_co_co_nest_random_functions_legally_used` is still called for all models (L178). The condition is only for synapse-specific CoCos. Otherwise, we get delay and weight variable asserts even for neuron models.